tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -0,0 +1,253 @@
+import argparse
+import time
+import math
+
+import pandas as pd
+import numpy as np
+
+import common
+import gmhazard_calc as gc
+from empirical.util import empirical_factory, classdef, openquake_wrapper_vectorized
+
+# fmt: off
+PERIOD = [0.01, 0.02, 0.03, 0.04, 0.05, 0.075, 0.1, 0.12, 0.15, 0.17, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.8,
+          0.9, 1.0, 1.25, 1.5, 2.0, 2.5, 3.0, 4.0, 5.0, 6.0, 7.5, 10.0]
+# fmt: on
+
+IM_TYPE_LIST = [
+    gc.im.IMType.PGA,
+    gc.im.IMType.pSA,
+    gc.im.IMType.Ds595,
+    gc.im.IMType.Ds575,
+    gc.im.IMType.AI,
+    gc.im.IMType.CAV,
+    gc.im.IMType.PGV,
+]
+
+MAG = [
+    5.25,
+    5.75,
+    6.25,
+    6.75,
+    7.25,
+    8.0,
+]  # Definitions for the maximum magnitude / rjb relation
+DIST = [125, 150, 175, 200, 250, 300]
+
+
+def get_rupture_context_df(
+    distance_store: gc.dbs.SiteSourceDB,
+    fault_df: pd.DataFrame,
+    site: str,
+    site_data: pd.Series,
+    nhm_data: pd.DataFrame,
+    tect_type: str,
+):
+    """Creating the form of dataframe for the vectorized OQ Wrapper
+    By combining site, distance and rupture information
+    """
+    distance_df = fault_df.merge(
+        distance_store.station_data(site), left_on="fault_name", right_index=True,
+    )
+    rupture_df = nhm_data.merge(
+        distance_df, left_on="fault_name", right_on="fault_name"
+    )
+    max_dist = np.minimum(
+        np.interp(rupture_df.mag, MAG, DIST),
+        common.get_max_dist_zfac_scaled(site_data),
+    )
+    rupture_df = rupture_df[rupture_df["rjb"] < max_dist]
+
+    return polish_rupture_context_df(rupture_df, site_data, tect_type)
+
+
+def polish_rupture_context_df(
+    rupture_context_df: pd.DataFrame, site_data: pd.Series, tect_type: str,
+):
+    """
+    Combining the given rupture_context_df(Distance and Rupture) with Site information
+    to create a OQ's RuptureContext like DF to be used with OQ's Vectorized Wrapper
+    """
+    # Adding missing columns
+    # Site Parameters
+    rupture_context_df["vs30"] = site_data.vs30
+    rupture_context_df["vs30measured"] = (
+        site_data.vs30measured if site_data.get("vs30measured") is not None else False
+    )
+    rupture_context_df["z1pt0"] = (
```

Comment:
These two can be simplified to 
`rupture_context_df["z1pt0"] = site_data.get("z1p0"]`

And then if OQ actually can't handle nan values, use fillna to convert nan to None, i.e.
`rupture_context_df[["z1...", "z2..."]] = `rupture_context_df[["z1...", "z2..."]].fillna(...)`

Reply:
Within the common module, get_work function does something like
```python
site_df = station_df.merge(vs30_df, left_index=True, right_index=True)

    # Combine Z-values and site data (if available)
    site_df = (
        site_df.join(formats.load_z_file(z_file)) if z_file is not None else site_df
    )
```
and while I am commenting on this, I noticed that to run calc_emp_ds, it must have z_file to pass anyway and site_df must have z values. then I can just do something like
`df = df.fillna(0)` instead and  just rename a few columns